CMakeLists.txt

```diff
@@ -42,6 +46,10 @@ else ()
     set(MMDEPLOY_LIB_TYPE STATIC)
 endif ()
 
+if (MMDEPLOY_BUILD_SDK_CSHARP_API)
```

Comment:
I think this condition is not necessary, since we already checked `if (MMDEPLOY_BUILD_SDK_CSHARP_API OR MMDEPLOY_BUILD_SDK_MONOLITHIC)` in `csrc/mmdeploy/apis/c/mmdeploy/CMakeLists.txt`

Reply:
But if we remove this condition, `add_library(mmdeploy SHARED)` in `csrc/mmdeploy/apis/c/mmdeploy/CMakeLists.txt` failed

Reply:
In csrc/mmdeploy/apis/c/mmdeploy, I only build OBJECT if MMDEPLOY_BUILD_SDK_MONOLITHIC is ON. If remove the condition,  there is no OBJECTS added to mmdeploy, that's why you failed.

Reply:
No matter what MMDEPLOY_BUILD_SDK_MONOLITHIC is, we can always build OBJECT targets. 
And when MMDEPLOY_BUILD_SDK_MONOLITHIC is ON, we build target `mmdeploy`